raft_paper_test.go

```diff
@@ -164,24 +156,17 @@ func testNonleaderStartElection(t *testing.T, state StateType) {
 	}
 	r.advanceMessagesAfterAppend()
 
-	if r.Term != 2 {
-		t.Errorf("term = %d, want 2", r.Term)
-	}
-	if r.state != StateCandidate {
-		t.Errorf("state = %s, want %s", r.state, StateCandidate)
-	}
-	if !r.trk.Votes[r.id] {
-		t.Errorf("vote for self = false, want true")
-	}
+	assert.Equal(t, uint64(2), r.Term)
+	assert.Equal(t, StateCandidate, r.state)
+	assert.True(t, r.trk.Votes[r.id])
+
 	msgs := r.readMessages()
 	sort.Sort(messageSlice(msgs))
```

Comment:
For another PR: consider removing `messageSlice`, and using `slices.SortFunc` with some stable comparator.

Reply:
Would this be what you had in mind?
```
slices.SortStableFunc(data, orderMessageAsc)
func orderMessageAsc(a pb.Message, b pb.Message) int {
	strA := fmt.Sprint(a)
	strB := fmt.Sprint(b)
	if strA > strB {
		return 1
	} else if strA < strB {
		return -1
	} else {
		return 0
	}
}
```